File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int length(ListNode* head){
        int cnt=0;
        ListNode *temp=head;
        while(temp){temp=temp->next;cnt++;}
        return cnt;
    }
    ListNode* rotateRight(ListNode* head, int k) {
        if(!head) return 0;
        int n=length(head);
        int rot=n-(k%n)-1;
         ListNode *temp=head,*p;
        while(rot--){
            temp=temp->next;
        }
        p=temp->next;
        temp->next=NULL;
       auto *root=p;
        while(p->next){
            p=p->next;
        }
        p->next=head;
        return root;
    }
};

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if(not head or not head.next or not k):
            return head
        
        Len=1
        curr,temp=head,head
        while(temp.next):
            temp=temp.next
            Len+=1
            
        temp.next=head
        
        k%=Len
        for i in range(Len-k-1):
            curr=curr.next
        new_head=curr.next
        curr.next=None
        return new_head
        
            
        
        